debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4

```diff
@@ -1717,8 +1717,8 @@ uninstallPlugin
 //    Set and show statements
 
 setStatement
-    : SET variableClause ('=' | ':=') expression
-      (',' variableClause ('=' | ':=') expression)*                             #setVariable
+    : SET variableClause ('=' | ':=') (expression | ON)
```

Comment:
It works, but wouldn't it make more sense to make `ON` part of `expression`? Or is it really _only_ allowed exactly here?

Reply:
I'm not MySQL expert to be able to say it for sure that this is the only place where it's allowed. And at the same time I'm not able to say for sure that adding it into `expression`, which is widely used, won't cause any issues as `ON` can have different meaning in some other circumstances/expressions. So this seems to me to be more safe change.

Reply:
Ok, makes sense, let's do it the proposed way then.

For some context, in general, we don't care about the semantics of these grammar parts at all from a Debezium perspective (just need to be able to parse it _somehow_, so we can properly parse what we're actually after, which is table DDL). So we can sometimes take shortcuts, though the aspiration still is to have a grammar that makes sense (not the least given we feed back those changes upstream to Antlr).

Reply:
Thanks for explanation - this is what I thought, that for Debezium it doesn't matter, at as I submitted this change also to Antlr, I don't want to break anything. Someone more experience can improve it in upstream if needed.